Match Custom Audio

gauravbhardwaj7 opened the pull request:
1.) Updated ShazamViewModel with functions buffer(), signature() and startMatching().
2.) Imported custom mp3 "ird" into resources and used it as audioFileURL in signature().

gauravbhardwaj7 committed:
Match Custom Audio

akshaaatt commented on Listenbrainz/ViewModel/ShazamViewModel.swift:
```diff
@@ -38,12 +38,18 @@ final class ShazamViewModel: NSObject, ObservableObject {
     private lazy var inputNode = self.audioEngine.inputNode
     private lazy var bus: AVAudioNodeBus = 0
+    @Published private(set) var isMatching = false
+    private let audioFileURL = Bundle.main.url(forResource: "ird", withExtension: "mp3")
```
@gauravbhardwaj7 we want to detect the mp3 files already present in the device of the user and then have them select which song they wanna match.

akshaaatt replied:
We can also allow the user to input(add) files and then save that file in the storage of our app and match then

gauravbhardwaj7 replied:
Alright!, will implement it:)

gauravbhardwaj7 replied:
implemented it, please have a look :)

akshaaatt reviewed (changes requested):
Thanks for the awesome work! 💯 

In order to complete this PR though, we need to incorporate the suggested feature.

gauravbhardwaj7 committed:
Updated ShazamViewModel

Updated ShazamViewModel and deleted previous audio resource.

gauravbhardwaj7 committed:
Added DocumentPicker

Added DocumentPicker for importing audio file and matching that audio file with the help of updated ShazamViewModel.
